homeassistant/components/accuweather/weather.py

```diff
@@ -66,19 +73,25 @@ def __init__(
     ) -> None:
         """Initialize."""
         super().__init__(coordinator)
-        self._unit_system = API_METRIC if coordinator.is_metric else API_IMPERIAL
-        wind_speed_unit = self.coordinator.data["Wind"]["Speed"][self._unit_system][
-            "Unit"
-        ]
-        if wind_speed_unit == "mi/h":
-            self._attr_wind_speed_unit = SPEED_MILES_PER_HOUR
+        # Coordinator data is used also for sensors which don't have units automatically
+        # converted, hence the weather entity's native units follow the configured unit
+        # system
+        if coordinator.is_metric:
+            self._attr_native_precipitation_unit = LENGTH_MILLIMETERS
+            self._attr_native_pressure_unit = PRESSURE_HPA
```

Comment:
Should we not just set it to metric or imperial as it would anyway be converted to users setting?

Reply:
Yes I did that first, but the problem is that the forecast data API response is in metric or imperial, and we want to reuse the data also for sensors.
We can't fetch data both for metric and imperical because if accuweather's extremely limited API rate for free developer access (50 per day).

Reply:
One suggestion would be to always collect one or the other and then use HA built-in conversions for presenting proper where "automatic" conversions don't apply but that is far bigger than this PR of course.